protos/index.proto

```diff
@@ -64,6 +64,12 @@ message PQ {
 
   // Number of sub vectors.
   uint32 num_sub_vectors = 2;
+
+  // Vector dimension
+  uint32 dimension = 3;
+
+  // Codebook. `dimension * 2 ^ num_bits` of float32s.
+  repeated float codebook = 4;
```

Comment:
For `1028D,  nbits = 8`, this can be 1MB. 

I guess I can make this into an array, but it requires 2 I/O (one for reading this protobuf, and the second one to read the actual codebook), which has data dependency so that we can not parallelize them.  But it gives a chance to read the array in parallel if the array is too big.